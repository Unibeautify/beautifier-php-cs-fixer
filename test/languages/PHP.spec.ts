import * as fs from "fs";
import * as path from "path";

import { newUnibeautify, Beautifier } from "unibeautify";
import beautifier from "../../src";
import { raw } from "../utils";

describe("should successfully beautify PHP files", () => {
  // tslint:disable:mocha-no-side-effect-code
  testFile("test1.php");
  testFile("test2.php");
  testFile("test3.php");
});

function testFile(fixtureFileName: string) {
  test(`should successfully beautify file ${fixtureFileName}`, () => {
    const text: string = fs
      .readFileSync(path.resolve(__dirname, `../fixtures/${fixtureFileName}`))
      .toString();
    const unibeautify = newUnibeautify();
    unibeautify.loadBeautifier(beautifier);
    return unibeautify
      .beautify({
        languageName: "PHP",
        options: {
          PHP: {
            indent_style: "space",
            indent_size: 2
          }
        },
        text
      })
      .then(results => {
        expect(raw(results)).toMatchSnapshot();
      });
  });
}
