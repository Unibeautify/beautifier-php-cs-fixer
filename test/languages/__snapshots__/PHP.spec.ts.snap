// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successfully beautify PHP files should successfully beautify file test1.php 1`] = `
<?php

$hello = "world";

if (!isset($_SESSION)) {
    session_start();
}
require_once("sql.req.php");


// friend requests
$q = $mysqli->query("");
$num = $q->num_rows;
echo '<a href="notifications.php">';
if ($num>0) {
    echo '<i class="fa fa-star white"></i>';
} else {
    echo '<i class="fa fa-star-o"></i>';
}
echo '</a>';


// new messages
$q = $mysqli->query("");
$num = $q->num_rows;
echo '<a href="messages.php">';
if ($num>0) {
    echo '<i class="fa fa-envelope white"></i>';
} else {
    echo '<i class="fa fa-envelope-o"></i>';
}
echo '</a>';

`;

exports[`should successfully beautify PHP files should successfully beautify file test2.php 1`] = `
<?php
function add($a, $b)
{
    return $a +$b;
}
echo   add(1, 2);

`;

exports[`should successfully beautify PHP files should successfully beautify file test3.php 1`] = `
<?php

class Date
{
    const FORMAT_FULL = 'full';
    const FORMAT_LONG = 'long';
    const FORMAT_MEDIUM = 'medium';
    const FORMAT_SHORT = 'short';

    public static function formatDate($time, $format)
    {
        if (!$time || $time == null) {
            return '';
        }

        return $time;
    }
}

`;
